model/ClientConfig.ts

```diff
@@ -1,3 +1,4 @@
+import { RetryOnStatusCode } from './HttpClient';
```

Comment:
I'm not sure it makes sense to import from an implementation class. See my suggestion for line 5.